### ISSUE TITLE
[Bug] Phone shows light theme on dark mode instead of dark theme on dark mode

### ISSUE THREAD

**Issue by 1chooo** [LelouchFR/skill-icons#171]
### Description of the bug

I check the icon on my iPhone all the icon is default in the light theme; however, in my laptop it’s default dark theme. Maybe we need to add configuration to make sure the icons display the dark mode successfully when using the mobile phone.

![IMG_5237](https://github.com/user-attachments/assets/e0bc39eb-263f-4f84-80cd-707e676599f7)

### Steps To Reproduce

1. Just view in the GitHub Mobile App with the dark theme 

### Code of Conduct Acknowledgment

- [X] I have read and agree to the Code of Conduct.

**Comment by 1chooo** [comment]
I also take the screenshot of your homepage, all the icons change into light mode.

![IMG_5238](https://github.com/user-attachments/assets/e9c2862f-042c-47d7-9b7a-96da4f722106)

**Comment by LelouchFR** [comment]
I've been aware of that, but I don't really have an idea of why :/ it may be because of the way that safari is rendering svgs and doesn't really take in count the dark theme and takes everything that is in the css for the theme and parses it as it without media query and shows up the last element changes that are there...

**Comment by 1chooo** [comment]
Hey @LelouchFR 

Is this issue caused by this logic? I noticed that only this part controls whether the theme is light, dark, or auto. Therefore, I believe the problem might be that "in phone mode, some icons default to auto and appear in light mode." Perhaps if we set the default icon to dark mode, we can resolve the issue.


https://github.com/LelouchFR/skill-icons/blob/b8b358af6ce3586e9097a16c20a33ef8be7a62ea/api/index.go#L264-L270

**Comment by 1chooo** [comment]
```md
![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=dark)
```

![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=dark)

**Comment by 1chooo** [comment]
> ```md
> 
> ![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=light)
> 
> ```
> 
> 
> 
> ![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=dark)

![image](https://github.com/user-attachments/assets/3679819a-f590-4070-b060-58c441d095b2)

**Comment by 1chooo** [comment]
> ```md
> 
> ![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=dark)
> 
> ```
> 
> 
> 
> ![My Skills](https://go-skill-icons.vercel.app/api/icons?i=java,kotlin,nodejs,figma&theme=dark)

![image](https://github.com/user-attachments/assets/9e7b4905-bf07-4447-997a-7ed4cec66dc7)

**Comment by LelouchFR** [comment]
> Hey @LelouchFR
> 
> Is this issue caused by this logic? I noticed that only this part controls whether the theme is light, dark, or auto. Therefore, I believe the problem might be that "in phone mode, some icons default to auto and appear in light mode." Perhaps if we set the default icon to dark mode, we can resolve the issue.
> 
> https://github.com/LelouchFR/skill-icons/blob/b8b358af6ce3586e9097a16c20a33ef8be7a62ea/api/index.go#L264-L270

Yes, but the point of the auto theme, is to be the default one, else, the problem is the same one, if the person has a github lightmode on their phone, the icons will be in dark...

As for the code, it only checks the files and if they have an "extension" like `-light`, `-dark` or `-auto`, if they do, they get stripped to get into the themedIcons variable
